Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -25,6 +25,19 @@ public IonObjectSerializer(IonSerializer ionSerializer, IonSerializationOptions
 
         public override object Deserialize(IIonReader reader)
         {
+            var ionConstructors = targetType.GetConstructors(BINDINGS).Where(IsIonConstructor).Take(2);
```

Comment:
Why do we only `Take(2)`

Reply:
`Take(2)` was added for a slight performance enhancement as we don't need more than 2 to run the following logic (since 2 or more IonConstructors result in an exception).